platform/darwin/src/MGLOpenGLStyleLayer.h

```diff
@@ -17,6 +17,7 @@ typedef struct MGLStyleLayerDrawingContext {
     CLLocationDirection direction;
     CGFloat pitch;
     CGFloat fieldOfView;
+    double projectionMatrix[16];
```

Comment:
As previously discussed in #4732, we should make this type either a `GLKMatrix4` or a `CATransform3D`. Otherwise, a plain `double[]` is a bit of a pain to convert to either `GLKMatrix4` or `CATransform3D` in Swift. Whichever of the two types we choose, it might be worth providing a [conversion function from one to the other](https://stackoverflow.com/a/23478943/4585461) in MGLGeometry.h.

Reply:
`GLKMatrix4` and `GLFloat` are lower precision than the `double` types required for projection. Using those types would provide less precise results.

Reply:
This would be a good argument for using `CATransform3D`, whose fields are `CGFloat`s, which are typedefs for `double` on 64-bit systems.